windpowerlib/data.py

```diff
@@ -187,45 +187,84 @@ def store_turbine_data_from_oedb(
         for index in turbine_data.index:
             if (
                 turbine_data["{}_wind_speeds".format(curve_type)][index]
                 and turbine_data["{}_values".format(curve_type)][index]
             ):
-                df = (
-                    pd.DataFrame(
-                        data=[
-                            eval(
-                                turbine_data[
-                                    "{}_wind_speeds".format(curve_type)
-                                ][index]
-                            ),
-                            eval(
-                                turbine_data["{}_values".format(curve_type)][
-                                    index
-                                ]
-                            ),
-                        ]
-                    )
-                    .transpose()
-                    .rename(
-                        columns={
-                            0: "wind_speed",
-                            1: turbine_data["turbine_type"][index],
-                        }
+                try:
+                    df = (
+                        pd.DataFrame(
+                            data=[
+                                eval(
+                                    turbine_data[
+                                        "{}_wind_speeds".format(curve_type)
+                                    ][index]
+                                ),
+                                eval(
+                                    turbine_data["{}_values".format(curve_type)][
+                                        index
+                                    ]
+                                ),
+                            ]
+                        )
+                        .transpose()
+                        .rename(
+                            columns={
+                                0: "wind_speed",
+                                1: turbine_data["turbine_type"][index],
+                            }
+                        )
                     )
+                    if not df.wind_speed.duplicated().any():
+                        curves_df = pd.merge(
+                            left=curves_df, right=df, how="outer", on="wind_speed"
+                        )
+                except:
+                    broken_turbine_data.append(turbine_data.loc[index, "turbine_type"])
+
+        # warning in case of broken turbine data
+        if len(broken_turbine_data) > 0:
+            issue_link = ("https://github.com/OpenEnergyPlatform/data-preprocessing"
+                          "/issues/28")
+            # in case only some data is faulty, only give out warning
+            if len(broken_turbine_data) < 0.2 * len(turbine_data):
+                logging.warning(
+                    f"The turbine library data contains faulty {curve_type}s. The "
+                    f"{curve_type} data can therefore not be loaded for the following "
```

Comment:
I am not sure, but I think that one broken data set can make the whole file unusable. Maybe we should add a hint that in that case it is possible to use the function `restore_default_turbine_data()`

Reply:
Actually I added these checks so that one broken data set could not make the newly downloaded data unusable, as happened in the last weeks due to bugs in the oedb turbine library data. I moved the checks to a separate function now to be able to test it independent of the oedb data and added tests to show that it is working correctly.